R/dgp-lib-utils.R

```diff
@@ -26,6 +27,40 @@
 #'   "Xtest" and "ytest".
 NULL
 
+#' Helper function to generate a coefficient vector.
+#' 
+#' @inheritParams shared_dgp_lib_args
+#' @param betas Coefficient vector. If a scalar is provided, the coefficient 
+#'   vector is a constant vector. If \code{NULL} (default), entries in the 
+#'   coefficient vector are drawn iid from N(0, \code{sd}^2).
+#' @param s Sparsity level. Coefficients corresponding to features after the 
+#'   \code{s}th position (i.e., positions i = \code{s} + 1, ..., \code{p}) are 
+#'   set to 0.
+#' @param sd (Optional) SD of normal distribution from which to draw 
+#'   \code{betas}. Only used if \code{betas} argument is \code{NULL}.
+#' @param betas_name Name of coefficient variable to use in error message.
+#' 
+#' @returns A vector of length \code{p}.
+#' 
+#' @keywords internal
+generate_coef <- function(betas = NULL, p, s = p, sd = 1, 
```

Comment:
I think this method (and perhaps others like it) might be useful to export, in case users want to reuse for their own DGPs.

Maybe we should give `p` a default value of 1?

Also, just a general comment on optional args: I tend to use a NULL default value, and I've seen you use the empty default value + `missing()` strategy a lot. I don't have a clear sense of which is better when, though if all else is equal I'd say the latter is probably a bit cleaner looking in the code and docs.

Reply:
We can definitely export this and add `p = 1`. 

Side note: I was thinking of maybe making this a more general function before exporting but haven't gotten the chance to do so. Specifically, I want to be allow users to be able to randomly generate coefficients from non-normal distributions (see issue #109).